src/Arcus.Messaging.Pumps.ServiceBus/Configuration/AzureServiceBusMessagePumpSettings.cs

```diff
@@ -24,6 +24,51 @@ public class AzureServiceBusMessagePumpSettings
         private readonly TokenCredential _tokenCredential;
         private readonly IServiceProvider _serviceProvider;
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AzureServiceBusMessagePumpSettings"/> class.
+        /// </summary>
+        /// <param name="entityName">The name of the entity to process.</param>
+        /// <param name="subscriptionName">The name of the subscription to process.</param>
+        /// <param name="serviceBusEntity">The entity type of the Azure Service Bus.</param>
+        /// <param name="getConnectionStringFromConfigurationFunc">The function to look up the connection string from the configuration.</param>
+        /// <param name="options">The options that influence the behavior of the <see cref="AzureServiceBusMessagePump"/>.</param>
+        /// <param name="serviceProvider">The collection of services to use during the lifetime of the <see cref="AzureServiceBusMessagePump"/>.</param>
+        /// <exception cref="ArgumentNullException">
+        /// Thrown when the <paramref name="getConnectionStringFromConfigurationFunc"/>, <paramref name="options"/> or <paramref name="serviceProvider"/> is <c>null</c>.
+        /// </exception>
+        public AzureServiceBusMessagePumpSettings(
+            string entityName,
+            string subscriptionName,
+            ServiceBusEntityType serviceBusEntity,
+            Func<IConfiguration, string> getConnectionStringFromConfigurationFunc,
```

Comment:
I'm not a fan of having to retrieve the connctionstring from config.
Why not just pass in the connectionstring as a string ?  Its the responsibility of the user to retrieve it from any location he wants.

Reply:
In other words, my idea would be to introduce a method that adds a messagepump and that accepts the connectionstring that must be used as a parameter:

```
public static ServiceBusMessageHandlerCollection AddServiceBusQueueMessagePump (
      this IServiceCollection services,
      string connectionString,
      string queueName,
      Action<IAzureServiceBusQueueMessagePumpOptions> options) {}
```
Maybe also provide a version that doesn't have the queueName parameter, because it's possible that the queue to which to connect is provided in the connectionstring as well.
(And for topics as well).

Reply:
Maybe an idea to have a `Func<IServiceProvider, ServiceBusClient>`? That would also make it possible to still use the secret store.

Reply:
Ok, I agree.
Just wondering why you would want to return a `ServiceBusClient` instead of a string (connectionstring).  Don't we expose the internal workings then ? And make it harder too adapt to change afterwards ?

Reply:
I would even opt to later move to use the `ServiceBusClient`s registered by the Azure SDK extensions (like `services.AddAzureClients(clients => clients.AddServiceBusClient(…))`) instead of us creating and managing these clients ourselves. So, having this delegate would already provide people with using those built-in clients already.
Exposing inner workings, that’s true. Although in this case it is not an inner working of our own creation (like if we would expose how custom message handlers are registered, or how message routing works), but a Microsoft known type (like if we would expose `TokenCredential`).
But, still true, it exposes a bit more than before.

Reply:
I've created something locally as a proposal, we can go over it separately to see if it fits with you had in mind.

Reply:
Yes, I'll schedule a meeting for it.

Reply:
Discussed this offline.
2 follow PR's will be created:
- one where we introduce the functionality to establish a connection by offering a factory that gives access to the ServiceProvider and is responsible for returning a `ServiceBusClient` instance.  Next to that, an extension method will be created that accepts a `TokenCredential` which is then used by Arcus.Messaging to establish the connection to ServiceBus
- another PR where we deprecate the functionality that allows the MessagePump to create its own servicebus-topic-subscription